src/main/java/duke/dependencies/executor/Executor.java

```diff
@@ -0,0 +1,152 @@
+package duke.dependencies.executor;
+
+
+import duke.dependencies.executable.CommandType;
+import duke.dependencies.executable.Executable;
+import duke.dependencies.storage.Storage;
+import duke.dependencies.storage.TaskList;
+import duke.dependencies.task.Task;
+
+
+import static duke.dependencies.executable.CommandType.*;
+
+/**
+ * Class that enables the logical manipulation of given Executable. Interprets the type of Command and
+ * carries out the command on the associated Task and the task list of the user.
+ *
+ */
+public class Executor {
+
+    private static final TaskList TASK_LIST = TaskList.initStorage();
+
+    private static final Storage STORAGE = new Storage();
+
+    /* Half-assed attempt at concurrency lock.
+    There should be no need for concurrency
+    in this application. */
+    private CommandType commandState;
+
+    /** Private constructor. */
+    private Executor() {}
+
+    /**
+     * Initializer for the executor. Returns the Executor object.
+     *
+     * @return Executor object.
+     */
+    public static Executor initExecutor() {
+        return new Executor();
+    }
+
+    /**
+     * Executes the given command and returns with a reply indicating the state of completion of given Executable.
+     *
+     * @param executable The command to be executed.
+     * @return String specifying what happened/what was done (no newline character at end of reply).
+     */
+    public String receiveAndExec(Executable executable) {
+        // TODO: Adding of new commands is to be done here.
+        switch(executable.getType()) {
+            case LIST:
+                setState(LIST);
+                break;
+
+            case ADD:
+                setState(ADD);
+                break;
+
+            case DELETE:
+                setState(DELETE);
+                break;
+
+            case DONE:
+                setState(DONE);
+                break;
+
+            case FIND:
+                setState(FIND);
+                break;
+
+            case CLEAR:
+                setState(CLEAR);
+                break;
+
+            default:
+                setState(INVALID);  // Should never reached this stage.
+                break;
+
```

Comment:
Another minor one, but the coding standard mentions that there's no need to indent for the `case` clauses.

Reply:
Agreed, no indentation needed for case clauses.